docs/rfds/session-info-update.mdx

```diff
@@ -0,0 +1,372 @@
+- **No breaking changes**: Existing agents and clients continue working
+- **Graceful degradation**: Clients that don't handle this notification simply ignore it
+- **No new capability needed**: Uses existing `session/update` infrastructure
+
+### Design Decisions
+
+**Why notification instead of request/response?**
+
+- Consistent with existing `SessionUpdate` patterns (`available_commands_update`, `current_mode_update`)
+- Agents initiate updates based on conversation state
+- Simpler than bidirectional request/response
+- Enables real-time updates without polling
+
+**Why make all fields optional?**
+
+- Allows partial updates (only update what changed)
+- More efficient - don't resend unchanged data
+- Flexible for different use cases
+- Mirrors partial update patterns in other protocols
+
+**Why not include `sessionId` and `cwd` in the update?**
+
+- `sessionId` is already in the notification params
+- `cwd` is immutable (set in `session/new`, never changes)
+- Keeps update focused on mutable metadata
+
+**How do `_meta` updates work?**
+
+- **Merge semantics**: New `_meta` fields are merged with existing ones
+- To clear a specific field: `"_meta": { "fieldName": null }`
+- To clear all custom metadata: `"_meta": null`
+
+### Security Considerations
+
+- **No additional security concerns**: Uses existing session authentication
+- **Input validation**:
+  - Agents should validate title length (recommend 500 chars max)
+  - Sanitize metadata to prevent injection
+  - Validate `_meta` structure based on agent requirements
+- **Resource limits**: Agents should limit update frequency and metadata size
+
+## Frequently asked questions
+
+### Why not create a new endpoint like `session/update-metadata`?
+
+The notification pattern is more appropriate because:
+
+1. **Consistency**: Other dynamic session properties (commands, modes) use notifications
+2. **Agent-initiated**: Agents typically generate titles from conversation context
+3. **Real-time**: No request/response overhead, updates flow naturally
+4. **Simpler**: Reuses existing `session/update` infrastructure
+
+### How does this work with `session/list`?
+
+The updated metadata should persist and be reflected in subsequent `session/list` calls. The notification provides real-time updates to connected clients, while `session/list` provides the current state for discovery.
+
+### Can clients trigger title updates?
+
+This RFD covers agent-initiated updates. Client-initiated updates could work by:
+
+1. Client sends a prompt asking to rename session
+2. Agent updates its internal state
+3. Agent sends `session_info_update` notification
+4. Client receives and displays the update
+
+A future RFD could add explicit request/response for this if needed.
+
+### What if multiple updates are sent in quick succession?
+
+Clients should apply updates incrementally in order. Each notification represents a delta, not a full replacement (except for fields explicitly set to `null`).
```

Comment:
We don't currently have this unset behavior in tool calls (probably should, but would be a different topic)
Just pointing out we would differ here, but could plan to align these in the future